app/models/concerns/cfe_submission_state_machine.rb

```diff
@@ -71,14 +72,20 @@ module CFESubmissionStateMachine
 
       event :irregular_income_created do
         transitions from: :other_income_created, to: :irregular_income_created, guard: :non_passported?
+        transitions from: :dependants_created, to: :irregular_income_created, guard: :non_passported?
```

Comment:
Many, if not all, of these transitions need to be removed, particularly if we implement an asynchronous calling CFE approach - which is being looked at by Colin.

Reply:
I add an integration test in the `spec/services/cfe/submission_manager_spec.rb` that covers this entire state machine flow, which would fail without this change. the `spec/concerns/cfe_submission_state_machine_spec.rb` unit specs are missing a lot transition event tests